realm/app-manager/src/manager.rs

```diff
@@ -180,6 +182,42 @@ impl Manager {
         Err(ProtocolError::SystemPowerActionFailed(format!("{:?}", e)))
     }
 
+    fn convert_sock_addr(storage: SockaddrStorage) -> Option<IpAddr> {
+        match storage.family() {
+            Some(AddressFamily::Inet) => storage.as_sockaddr_in().map(|addr| IpAddr::V4(addr.ip())),
+            Some(AddressFamily::Inet6) => {
+                storage.as_sockaddr_in6().map(|addr| IpAddr::V6(addr.ip()))
+            }
+            _ => None,
+        }
+    }
+
+    fn read_if_addrs() -> ProtocolResult<HashMap<String, NetAddr>> {
+        let ifaddrs = block_in_place(|| nix::ifaddrs::getifaddrs()).map_err(|e| {
+            error!("Failed to read ip addresses of interfaces: {:?}", e);
+            ProtocolError::GetIfAddrsError(format!("{:?}", e))
+        })?;
+
+        let mut net_addrs = HashMap::new();
+
+        for ifaddr in ifaddrs {
+            let ipaddr = ifaddr.address.map(Self::convert_sock_addr).flatten();
+
+            if let Some(addr) = ipaddr {
+                net_addrs.insert(
+                    ifaddr.interface_name,
+                    NetAddr {
+                        address: addr,
+                        netmask: ifaddr.netmask.map(Self::convert_sock_addr).flatten(),
+                        destination: ifaddr.destination.map(Self::convert_sock_addr).flatten(),
+                    },
+                );
+            }
+        }
+
+        Ok(net_addrs)
+    }
```

Comment:
Maybe we could move this functions outside the manager file? I suggest to create _net.rs_ file in utils subfolder and put them there. What do you think?

Reply:
done